### PR TITLE
Implementation of Gumbel Distribution

### PR DESCRIPTION
**Pull request by vishwakftw**
This closes https://github.com/probtorch/pytorch/issues/70.

### CODE REVIEW

**Comment by vishwakftw** [comment]
Oops, I think the master at probtorch/pytorch wasn't updated in quite a while.

**Comment by vishwakftw** [comment]
I think the last three are the commits required.

File: torch/distributions/gumbel.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    def __init__(self, loc, scale):
+        self.loc, self.scale = broadcast_all(loc, scale)
+        if isinstance(loc, Number) and isinstance(scale, Number):
+            batch_shape = torch.Size()
+        else:
+            batch_shape = self.scale.size()
+        super(Gumbel, self).__init__(batch_shape)
+
+    @constraints.dependent_property
```
Comment (fritzo) [inline torch/distributions/gumbel.py]: This constraint is not dependent on parameters, so instead you should simplify and move this up to
```py
params = ...
support = constraints.real
```

**Comment by vishwakftw** [comment]
I have opened this on `pytorch/pytorch`.

**Pull request closed without merging**